Slim/Control/Queries.pm

```diff
@@ -5736,7 +5805,7 @@ sub _getTagDataForTracks {
 	$tags =~ /e/ && do { $c->{'tracks.album'} = 1 };
 	$tags =~ /E/ && do { $c->{'tracks.extid'} = 1 };
 	$tags =~ /d/ && do { $c->{'tracks.secs'} = 1 };
-	$tags =~ /t/ && do { $c->{'tracks.tracknum'} = 1 };
+	$tags =~ /t/ && !$args->{workId} && do { $c->{'tracks.tracknum'} = 1 };
```

Comment:
Another UI decision. If the UI doesn't want track numbers, don't ask for them. If they ask for track numbers, return them. Such exceptions will bite us in the back if we let them sneak in.

Reply:
Another consequence of now having "sub-albums" (ie "album-work") as described above. When showing these, track numbers are superfluous and confusing. It's not a decision the UI should make, it's predicated on the nature of the track list returned - is it the full album or a sub-album?

Reply:
But it's up to the UI to display these track numbers or not, isn't it? Really: if the request is to return the track number, then we return the track number. If something doesn't want it, then make it not request it. This kind of mixing conditions is asking for trouble. Might be easier in the short term, but will lead to more confusion and inconsistent behaviour in the use case we don't know yet.

Reply:
> But it's up to the UI to display these track numbers or not, isn't it?

No. When calling this for a Work rather than a full album, track numbers are meaningless because the parent entity is the work, not the album.

If I had completely divorced Work from Album by creating new queries "workPerformances" and "workPerformanceTracks" (the latter would _never_ return a track number) I suppose it would be clearer, but also there would be massive code duplication.

Instead of the above code, I could remove the tag 't' if work_id is passed, at the top of the subroutine, with a line of explanation. If you prefer.

Reply:
No, fix the caller. Where would this be called from with both a `work_id` and the `t` tag?

Reply:
OK. @CDrummond , please don't send the "t" tag when invoking the titles command for a work (ie when passing work_id)

Reply:
@CDrummond this isn't going to work, because tracksQuery adds "t" to the tags depending on the sort. You would need to manually suppress tracknum if work_id is present, as I will have to do in BrowseLibrary for Default Skin.

But, @michaelherger , this is getting complicated: my previous suggestion of removing tag "t" in the query if work_id is present wouldn't be the first time a query manipulated the tags, as demonstrated by what the sort parameter does.  Are you sure we can't do it that way?

Reply:
@darrell-k I've updated Material to now show track numbers if work_id is present.

Reply:
@darrell-k Yeah, typo - I meant _not_ not _now_